enhancements/monitoring/crd-based-cmo.md

```diff
@@ -0,0 +1,149 @@
+---
+title: CRD Based CMO
+authors:
+  - "@marioferh"
+  - "@danielmellado"
+reviewers:
+approvers:
+creation-date: 2024-04-26
+last-updated: 2024-04-26
+status: provisional
+---
+
+# CRD Based CMO
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+* Currently, the monitoring stack is configured using a configmap. In OpenShift though the best practice is to configure operators using custom resources.
+
+
+## Motivation
+
+* The specification is well known and to a degree self-documenting
+* We can specify validation and legal values right in the CRD.
+* The APIServer will validate user resources based on our specifications, so users get immediate feedback on errors instead of having to check if their config was applied and check logs.
+* Many users expect to interact with operators through a CRD
+* Compatible with GitOps workflows. 
+* We can add [cross]validation rules to CRD fields to avoid misconfigurations
+* End users get a much faster feedback loop. No more applying the config and scanning logs if things don't look right. The API server will give immediate feedback
+* Organizational users (such as ACM) can manage a single resource and observe its status
+
+
+### Goals
+
+- Replace configmaps with CRD
+- Smooth transition for users
+
+## Proposal
+
+## Design Details
+
+To initiate the process, let's establish a feature gate that will serve as the entry point for implementing a CRD configuration approach. This strategy enables us to make incremental advancements without the immediate burden of achieving complete feature equivalence with the config map. We can commence with a the basics and progressively incorporate additional functionalities as they develop.
+
+One proposal for a minimal DoD was:
+    - Feature gate
+    - CRD Initial dev https://github.com/openshift/cluster-monitoring-operator/pull/2347/
+        Add controller-gen logic to makefile
```

Comment:
Working on the CRD on the CMO repo is probably good, but ultimately we'll want to api to live in `openshift/[api,client-go]`. This will also take care the generating the manifests, so probably best to not invest too much into the controler-gen logic.

Reply:
then, should we add right now the api to and client go to openshift/ ?

Reply:
sure, opening a PR there